### PR TITLE
feat: kill and stable RPCs [DTSS-595]

### PR DESCRIPTION
**Pull request by george-e-shaw-iv**
**What this PR does / why we need it**: 
* Adds a kill RPC and a stable RPC. The kill RPC allows the localizer server to be killed (best effort) by a client that spawns it. The stable RPC denotes whether or not the port-forwarder has seen any action in the last 2 seconds. This is a best-effort flag that the initial queue has been drained and that localizer is ready to be used by say, e2e tests.

**JIRA ID**: [DTSS-595]

**Notes for your reviewer**:

[DTSS-595]: https://outreach-io.atlassian.net/browse/DTSS-595

### CODE REVIEW

File: cmd/localizer/list.go [hunk 1]
```diff
@@ -44,12 +44,13 @@ func NewListCommand(_ logrus.FieldLogger) *cli.Command { //nolint:funlen
 			ctx, cancel := context.WithTimeout(c.Context, 30*time.Second)
 			defer cancel()
 
-			client, err := localizer.Connect(ctx, grpc.WithBlock(), grpc.WithInsecure())
+			client, closer, err := localizer.Connect(ctx, grpc.WithBlock(), grpc.WithInsecure())
 			if err != nil {
 				return errors.Wrap(err, "failed to connect to localizer daemon")
 			}
+			defer closer() //nolint:errcheck // Why: Nothing we can really do about the error that comes from closing the gRPC client connection.
```
Comment (deregtd) [inline cmd/localizer/list.go:51]: can you just have closer be a empty func or something if it returns an err?  i don't know what's actually idiomatic here, but it feels like it'd be safer to always do a defer closer() if possible.  but obvs you know go better than i do.

Comment (george-e-shaw-iv) [reply cmd/localizer/list.go:51]: yeah we can do that. the error is just from closing a grpc.ClientConn, so if that fails we can't do anything about it. I'll just eat the error in the closing function as opposed to having to nolint it everywhere

File: internal/server/grpc.go [hunk 2]
```diff
@@ -56,12 +56,13 @@ func (g *GRPCService) CleanupPreviousInstance(ctx context.Context, log logrus.Fi
 	defer cancel()
 
 	log.Info("checking if an instance of localizer is already running")
-	client, err := localizer.Connect(ctx, grpc.WithBlock(), grpc.WithInsecure())
+	client, closer, err := localizer.Connect(ctx, grpc.WithBlock(), grpc.WithInsecure())
+	defer closer() //nolint:errcheck // Why: Nothing we can really do about the error that comes from closing the gRPC client connection.
```
Comment (deregtd) [inline internal/server/grpc.go:60]: this one comes before the early exit :(

**Review by deregtd — APPROVED** [review]

File: internal/server/grpc.go (hunk 2 above)
Comment (george-e-shaw-iv) [reply internal/server/grpc.go:60]: good catch, that function could be nil.

**Pull request merged**